### PR TITLE
Prepare sodiumoxide for release

### PR DESCRIPTION
**Pull request by tasn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+
+* Update prebuilt windows binaries to those published at 2020-06-30 at libsodium.org
+* Update libsodium submodule to latest stable branch commit. (#420, #400)
+* Avoid ptr_cast feature as it was not yet stable in Rust 1.36. (#408)
+* Fix build script by using the correct arguments for xcopy (#407)
+* Avoid dev dependencies on tar and libflate by including libsodium as a Git submodule. (#393)
+* Add bindings to KDF functions (#351)
+* Expose libsodium's hexadecimal encoding/decoding helper functions (#390)
+* Expose libsodium's Base64 encoding/decoding helper functions (#391)
+* remove ill-advised gen_nonce functions
```
Comment (kpp) [inline CHANGELOG.md:11]:
```suggestion
* remove ill-advised gen_nonce functions (#395)
```

**Comment by tasn** [comment]
Fixed, thanks.

**Comment by kpp** [comment]
Update libsodium-sys/Cargo.toml too

**Comment by tasn** [comment]
Done.

**Comment by dnaq** [comment]
I’ll try to make the release on Tuesday/Wednesday.

**Comment by tasn** [comment]
@dnaq, great thanks a lot.

@kpp, obviously thanks to you too!

**Comment by dnaq** [comment]
@tasn can you investigate why the tests fail on Windows?

**Comment by tasn** [comment]
Hey, I unfortunately can't. I'm not a rust expert, nor do I know this library very well... I have no idea where to start.
All I changed in this PR is the version info and the changelog, so it's unlikely it introduced it.

**Comment by dnaq** [comment]
Np. @kpp could you take a look at it. I don't have access to a windows system to reproduce.

**Comment by kpp** [comment]
I don't know why it's not working...

**Comment by dnaq** [comment]
Seems a bit weird since all tests have passed on master. Might be some issue with either the appveyor conf or appveyor itself. I would like to see the tests run on a windows machine before merging though.

**Comment by dnaq** [comment]
So, I did some experimentation with the appveyor configuration here: #429 and saw that if we switched to using the prebuilt libraries on 64 bit msvc the tests passed. However there should be no reason for the tests failing when not using prebuilt libraries as well, so something weird is going on.

Also when looking through the tests it seems strange that 32-bit msvc and mingw used the prebuilt libraries, and so did 64-bit mingw, but 64-bit msvc wanted to use vcpkg. Seems like those tests have gone a bit stale. We should of course run all tests on all variants. So if someone wants to take care of that I would be glad. Or should we remove the vcpkg support completely since we haven't tested it for most build variants?, that should simplify the build.rs as well.

@kpp: what do you think?
@tasn: would you update the pull request with the changes in #429?

**Comment by dnaq** [comment]
Oh, and @kpp, maybe it would be easier to switch to azure pipelines in the future so we can run all tests there instead of supporting both travis and appveyor?

**Comment by tasn** [comment]
Done.

**Comment by Dylan-DPC** [comment]
@dnaq yeah that's the plan just that none of us have sat to do it

**Comment by kpp** [comment]
> Or should we remove the vcpkg support completely since we haven't tested it for most build variants?, that should simplify the build.rs as well.

I think we should remove the vcpkg support. IIRC it came out of box from some PR. But unfortunately it is broken, so we'd better get rid of it until somebody requests for its support.

**Comment by dnaq** [comment]
I added some more changes to #429, we can either merge that one or this one if you add the changes to it. Would feel more fair to merge this one and close #429

**Comment by tasn** [comment]
I updated this PR with your changes.

**Comment by dnaq** [comment]
Would you mind rebasing to clean up the history a bit? One commit for the vcpkg changes and one for the change log/version number bump?

**Comment by tasn** [comment]
Done.

**Comment by dnaq** [comment]
I still see four different commits, could you squash them into two?

**Comment by kpp** [comment]
Oh. Maybe lets merge a separate PR with vcpkg removal and then rebase this PR onto latest master?

**Comment by tasn** [comment]
Done.

**Comment by dnaq** [comment]
@tasn can you change the title and commit message of the vcpkg commit. @kpp I'm OK with both changes being in the same pull request as long as it's two different commits.

**Comment by tasn** [comment]
@dnaq, I can't guess what you have in mind... Could you please just do your changes in a branch (like the draft PR you did) and let me know once ready so I can update this PR?

Thanks!

**Comment by tasn** [comment]
Pushed, thanks!

**Pull request merged**